### PR TITLE
Fix a few things about tasks from the redesign

### PR DESCRIPTION
**Pull request by jasperhuangg**
### Details
- This didn't happen on staging or production, but for whatever reason the dev build of RN makes it so that multiline text inputs don't focus to the end. This PR fixes that.
- Makes it so the task title input in task creation accepts multiple lines
- Makes it so clicking the task header doesn't open the task assignee selector modal. Instead it now functions like threads, navigating you back to the parent report.

### Fixed Issues

$ https://github.com/Expensify/App/issues/22494
$ https://github.com/Expensify/App/issues/22495
$ https://github.com/Expensify/App/issues/22498

### Tests/QA
1. FAB > Assign Task.
2. Enter a large amount of text into the task title `This is a very very long task title that should occupy multiple lines, at the very least.`, verify that it can occupies multiple lines.
3. Finish creating the task with any other values for the other fields.
4. In the Task View, click the title. Verify that the cursor is at the end of the text input.
5. Close the task title modal. Click the header. Verify it takes you back to the task's parent report, and doesn't open the task assignee selector modal.

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
Had some issues with Android/iOS simulators, but was able to test on them prior and confirmed they were working
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/31285285/151164d5-e0fe-49d6-b6af-80eb3096c930

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/31285285/c9034581-d090-4c25-9a8e-31b7b2f9a841

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/31285285/9775c384-2b6b-471a-bf66-47e968fa72c4

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/31285285/98c06f72-3223-448f-bf2c-7d975d745557

</details>

<details>
<summary>iOS</summary>

</details>

<details>
<summary>Android</summary>

</details>

### CODE REVIEW

**Comment by Santhosh-Sellavel** [comment]
@jasperhuangg Conflicts please resolve

**Comment by jasperhuangg** [comment]
@Santhosh-Sellavel had to incorporate a few more changes into this PR, give me a moment while I update the screen recordings

**Review by thienlnam — COMMENTED** [review]
Appreciate the PR, thanks for fixing these things!

**Comment by jasperhuangg** [comment]
@Santhosh-Sellavel this is ready for review!

**Comment by Santhosh-Sellavel** [comment]
@jasperhuangg Are you gonna handle [this](https://github.com/Expensify/App/pull/22586#issuecomment-1630220208) or not?

**Comment by jasperhuangg** [comment]
@Santhosh-Sellavel fixed!

File: src/pages/home/HeaderView.js [hunk 1]
```diff
@@ -162,7 +162,7 @@ function HeaderView(props) {
                 {Boolean(props.report && title) && (
                     <View style={[styles.flex1, styles.flexRow, styles.alignItemsCenter, styles.justifyContentBetween]}>
                         <PressableWithoutFeedback
-                            onPress={() => (isTaskReport ? Navigation.navigate(ROUTES.getTaskReportAssigneeRoute(props.report.reportID)) : ReportUtils.navigateToDetailsPage(props.report))}
+                            onPress={() => !isTaskReport && ReportUtils.navigateToDetailsPage(props.report)}
```
Comment (thienlnam) [inline src/pages/home/HeaderView.js:165]: You can remove the !isTaskReport && here

**Review by thienlnam — CHANGES_REQUESTED** [review]

**Comment by Santhosh-Sellavel** [comment]
@jasperhuangg Do test steps Cover everything that needs to be tested, if not please update!

**Comment by Santhosh-Sellavel** [comment]
@jasperhuangg Unable to mark the task done.

1. Login as User A
2. FAB -> Assign Task
3. Enter task details 
4. Fill in other Pick assignees same logged-in user.
5. In the Task View Report try to click Mark as done. Unable to click it

https://github.com/Expensify/App/assets/85645967/4479a984-9371-4a86-a5db-0edeb51d5bb5

Screenshot 2023-07-12 at 2 05 02 AM

**Comment by jasperhuangg** [comment]
Woah, didn't think that could happen from my changes, will investigate

**Comment by Santhosh-Sellavel** [comment]
@jasperhuangg Not sure what went wrong earlier, it's not able to reproduce now!

**Comment by jasperhuangg** [comment]
@Santhosh-Sellavel nope, can't reproduce on my side. That looks an error thrown by Onyx when a key in Onyx doesn't start with the expected prefix. Perhaps you just had some bad Onyx data? Were you testing against staging, and did you create a brand new task report, or use an existing one?

**Comment by jasperhuangg** [comment]
@Santhosh-Sellavel are you able to help review this?

**Comment by Santhosh-Sellavel** [comment]
Yes, I'm running into weird issues. Keep getting these warnings.

image

**Comment by Santhosh-Sellavel** [comment]
@jasperhuangg

> Click the header. Verify it takes you back to the task's parent report and doesn't open the task assignee selector modal.

This fails

https://github.com/Expensify/App/assets/85645967/431ad48d-4f4c-415f-a29c-ff2f41699e9d

**Comment by jasperhuangg** [comment]
@Santhosh-Sellavel few questions in your Slack DM

**Comment by Santhosh-Sellavel** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web </summary>

https://github.com/Expensify/App/assets/85645967/ab732cff-77ce-4992-8bdc-d9072f4ee8eb

</details>

<details>
<summary>Mobile Web - Chrome & Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/85645967/a4108c14-a9c7-4ad8-90b5-d7fb91a704b6

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/85645967/cc39d767-69b2-402c-8f7e-864c4cc3736b

</details>

<details>
<summary>iOS & Android</summary>

https://github.com/Expensify/App/assets/85645967/9c05906a-9cf8-47d6-9422-2faaa1f055f2

</details>

**Comment by Santhosh-Sellavel** [comment]
All looks good. Just a couple of questions. 

1. While online, click the header near the subtitle that has the text `from some email` and take to the report where the task was shared. And click the task title in the header takes you to the details page of the chat where it was shared. Is it the expected behaviour?

2. While offline, after performing the steps in the test create a new task. When I click the `from some email` takes me to the chat as it was shared. But when clicking on the task name in the header takes me to details page and it is empty

Refer video for more understanding, can you answer those questions please thanks!

https://github.com/Expensify/App/assets/85645967/a33fb941-0103-4e80-a4d3-0f680511e76c

**Comment by jasperhuangg** [comment]
> All looks good. Just a couple of questions. 
> 
> 
> 
> 1. While online, click the header near the subtitle that has the text `from some email` and take to the report where the task was shared. And click the task title in the header takes you to the details page of the chat where it was shared. Is it the expected behavior?
> 
> 
> 
> 2. While offline, after performing the steps in the test create a new task. When I click the `from some email` takes me to the chat as it was shared. But when clicking on the task name in the header takes me to the details page and it is empty. Is it expected or what should we do now?
> 
> 
> 
> Refer video for more understanding, can you answer those questions please thanks!
> 
> 
> 
> https://github.com/Expensify/App/assets/85645967/a33fb941-0103-4e80-a4d3-0f680511e76c
> 
> 
> 
> 

1. Yes that's expected
2. I’m honestly not sure, I don't think that's expected? But this bug seems like it has to do with an existing issue that's we're fixing internally, so I wouldn't worry about it.

**Comment by Santhosh-Sellavel** [comment]
> But this bug seems like it has to do with an existing issue that's we're fixing internally, so I wouldn't worry about it.

Should I report it then?

**Review by Santhosh-Sellavel — APPROVED** [review]
LGTM tests well!
C+ Reviewed
🎀 👀 🎀

**Review by thienlnam — APPROVED** [review]
Going to go ahead and handle the review for this since I've been following along

**Pull request merged**